msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFFileExportMethod.java

```diff
@@ -77,84 +95,71 @@ public Void execute() throws MSDKException {
       for (int i = 0; i < scanStartPositions.length - 1; i++)
         scanIndexArray.setInt(i, scanStartPositions[i]);
 
-      if (canceled) {
-        writer.close();
-        return null;
-      }
-
       // Write the scan indices
+      // Create a variable and set the dimension scan_number=totalScans to it
       Dimension scanNumberDim = writer.addDimension(null, "scan_number", totalScans);
       List<Dimension> scanIndexDims = new ArrayList<>();
       scanIndexDims.add(scanNumberDim);
-
-      // Create scan index varaible
       Variable scanIndexVariable =
           writer.addVariable(null, "scan_index", DataType.INT, scanIndexDims);
 
-      progress = 0.25f;
+      progress = 0.2f;
 
       // data values storage dimension
+      // Dimension to store the data arrays of various scans
       Dimension pointNumDim =
           writer.addDimension(null, "point_number", scanStartPositions[totalScans]);
       List<Dimension> pointNumValDims = new ArrayList<>();
       pointNumValDims.add(pointNumDim);
 
-      // populate the mass values
+      // populate the mass and the intensity values
       Array massValueArray =
           Array.factory(double.class, new int[] {scanStartPositions[totalScans]});
+      Array intensityValueArray =
+          Array.factory(double.class, new int[] {scanStartPositions[totalScans]});
+
       idx = 0;
       for (MsScan scan : scans) {
         double mzValues[] = scan.getMzValues();
+        float intensityValues[] = scan.getIntensityValues();
```

Comment:
Each time you cal scan.getMzValues() or getIntensityValues() a new array is created and has to be collected by garbage collector later. This is exactly why we added the option to reuse the arrays:
````
double mzValues[] = new double[10000];
double intensityValues[] = new double[10000];
for ... {
mzValues= scan.getMzValues(mzValues);
intensityValues = scan.getIntensityValues(intensityValues);
}
````